examples/camera.rs

```diff
@@ -1,23 +1,113 @@
+use std::collections::HashSet;
+
 use macroquad::*;
 
+fn short_angle_dist(a0: f32, a1: f32) -> f32 {
+    let max = 360.0;
+    let da = (a1 - a0) % max;
+    2.0 * da % max - da
+}
+
+fn angle_lerp(a0: f32, a1: f32, t: f32) -> f32 {
+    a0 + short_angle_dist(a0, a1) * t
+}
+
+fn draw_cross(x: f32, y: f32, color: Color) {
+    let size = 0.1;
+    let thickness= 0.005;
+    draw_line(x - size, y, x + size, y, thickness, color);
+    draw_line(x, y - size, x, y + size, thickness, color);
+}
+
 #[macroquad::main("Camera")]
 async fn main() {
-    loop {
-        clear_background(RED);
+    let mut down_keys: HashSet<KeyCode> = HashSet::new();
 
-        // Render some primitives in camera space
+    let mut target = (0., 0.);
+    let mut zoom = 1.0;
+    let mut rotation = 0.0;
+    let mut smooth_rotation: f32 = 0.0;
+    let mut offset = (0., 0.);
+
+    'main: loop {
+        for &key in get_down_keys(&mut down_keys).iter() {
+            match key {
+                KeyCode::W => target.1 -= 0.1,
+                KeyCode::S => target.1 += 0.1,
+                KeyCode::A => target.0 += 0.1,
+                KeyCode::D => target.0 -= 0.1,
+                KeyCode::Left => offset.0 -= 0.1,
+                KeyCode::Right => offset.0 += 0.1,
+                KeyCode::Up => offset.1 += 0.1,
+                KeyCode::Down => offset.1 -= 0.1,
+                #[cfg(not(target_arch = "wasm32"))]
+                KeyCode::Q | KeyCode::Escape => break 'main,
+                _ => ()
+            }
+        }
+
+        match mouse_wheel() {
+            (_x, y) if y != 0.0 => {
+                // Normalise mouse wheel values is browser (chromium: 53, firefox: 3)
```

Comment:
Maybe do normalization on macroquad side, not on the user code side? 
Not necessarily as a part of this PR though

Reply:
Maybe move normalization in [mouse_wheel](https://github.com/not-fl3/macroquad/blob/master/src/lib.rs#L345) function?